lib/improver/tests/threshold/test_BasicThreshold.py

```diff
@@ -411,6 +422,26 @@ def test_threshold_below_fuzzy_miss(self):
             1, 1, 5, 5)
         self.assertArrayAlmostEqual(result.data, expected_result_array)
 
+    def test_threshold_unit_conversion(self):
+        """Test data are correctly thresholded with different units"""
+        expected_result_array = np.zeros((2, 5, 5))
+        expected_result_array[0][2][2] = 1.
```

Comment:
While I see your point, there are a couple of reasons I'd actually rather keep it as is in this case.  This is mostly for consistency with the other tests in this module, all of which are laid out in this non-verbose way, and partly so that I can re-use things like coordinates in setUp().

I do think both the plugin and tests in this case could do with some refactoring - the `Test_process` class is very large.  I don't know whether this is enough of an issue to raise as a follow on ticket for wider code consistency / maintainability?  But TBH if I did this one test I'd rather refactor them all, and that would be quite a lot of work (at least enough for a separate PR!).

Reply:
Yes, I take your point regarding consistency with the existing tests and I think that there might be scope for a follow on refactoring ticket but refactoring all the tests would be out of scope for this PR. Generally, I prefer for the unit tests to be human-readable, whilst I feel that this non-verbose style is not very human-readable. For this test, I'd be satisfied to leave the tests as they are.